### PR TITLE
Feature/cu p1btku boss basic structure

### PR DESCRIPTION
**Pull request by Mefiso**

### CODE REVIEW

**Review by Jordi-Gil — APPROVED** [review]

**Review by JoanMarcBardes — APPROVED** [review]

File: Penteract/Assets/Scripts/VideoSceneEnd.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    MEMBER(MemberType::SCENE_RESOURCE_UID, winSceneUID),
+
+};
+
+GENERATE_BODY_IMPL(VideoSceneEnd);
+
+void VideoSceneEnd::Start() {
+    // Set up video
+    componentVideo = GetOwner().GetComponent<ComponentVideo>();
+    componentVideo->SetVideoFrameSize(static_cast<int>(Screen::GetResolution().x), static_cast<int>(Screen::GetResolution().y));
```
Comment (GBMiro) [inline Penteract/Assets/Scripts/VideoSceneEnd.cpp:17]: Check componentVideo nullptr

File: Penteract/Assets/Scripts/VideoSceneEnd.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    parent = GetOwner().GetParent();
+
+}
+
+void VideoSceneEnd::Update() {
+
+    if ((componentVideo->HasVideoFinished() && componentVideo->IsActive()) || Input::GetKeyCodeDown(Input::KEYCODE::KEY_ESCAPE)) {
```
Comment (GBMiro) [inline Penteract/Assets/Scripts/VideoSceneEnd.cpp:24]: Same

File: Penteract/Assets/Scripts/AIDuke.cpp [hunk 3]
```diff
@@ -429,133 +520,112 @@ void AIDuke::OnAnimationEvent(StateMachineEnum stateMachineEnum, const char* eve
...
-		//if (animation->GetCurrentState()) animation->SendTrigger(animation->GetCurrentState()->name + "Hurt");
+		dukeCharacter.compAnimation->SendTrigger(dukeCharacter.compAnimation->GetCurrentState()->name + dukeCharacter.animationStates[Duke::DUKE_ANIMATION_STATES::PUSHED]);
+		dukeCharacter.StopShooting();
 		CalculatePushBackRealDistance();
+
+		OnShieldInterrupted();
+
 		// Damage
 		if (playerController->playerOnimaru.level2Upgrade) {
 			dukeCharacter.GetHit(playerController->playerOnimaru.blastDamage + playerController->GetOverPowerMode());
```
Comment (GBMiro) [inline Penteract/Assets/Scripts/AIDuke.cpp:607]: hitTaken = true needs to be add after GetHit

File: Penteract/Assets/Scripts/Duke.cpp [hunk 4]
```diff
@@ -49,132 +68,273 @@ void Duke::Init(UID dukeUID, UID playerUID, UID bulletUID, UID barrelUID, UID ch
...
 		}
 	}
 }
 
-void Duke::BulletHell()
-{
+void Duke::BulletHell() {
 	Debug::Log("Bullet hell");
+	if (attackDronesController) {
+		compAnimation->SendTrigger(compAnimation->GetCurrentState()->name + animationStates[static_cast<int>(DUKE_ANIMATION_STATES::PDA)]);
```
Comment (GBMiro) [inline Penteract/Assets/Scripts/Duke.cpp:110]: Is compAnimation being check if nullptr when calling it?

File: Penteract/Assets/Scripts/Duke.cpp [hunk 5]
```diff
@@ -49,132 +68,273 @@ void Duke::Init(UID dukeUID, UID playerUID, UID bulletUID, UID barrelUID, UID ch
...
+	}
 }
 
 void Duke::UpdateCharge(bool forceStop)
 {
+
+	if (trackingChargeTarget) {
+		float3 dir = player->GetComponent<ComponentTransform>()->GetGlobalPosition() - dukeTransform->GetGlobalPosition();
+		dir.y = 0.0f;
+		movementScript->Orientate(dir);
```
Comment (GBMiro) [inline Penteract/Assets/Scripts/Duke.cpp:149]: Check movementScript nullptr

File: Penteract/Assets/Scripts/AIDuke.cpp (hunk 3 above)
Comment (Mefiso) [reply Penteract/Assets/Scripts/AIDuke.cpp:610]: but there's no hitTaken here?

**Review by magalenyo — APPROVED** [review]
Great work

**Pull request merged**